libnopegl/src/node_colorstats.c

```diff
@@ -416,7 +418,13 @@ static void colorstats_draw(struct ngl_node *node)
     ngli_pipeline_compat_dispatch(s->init.pipeline_compat, s->init.wg_count, 1, 1);
 
     /* Waveform */
-    ngli_pipeline_compat_update_texture_info(s->waveform.pipeline_compat, s->waveform.texture_info);
+    const struct image *image = s->waveform.texture_info->image;
+    ngli_assert(image);
+    if (s->waveform.texture_image_rev != image->rev) {
+        ngli_pipeline_compat_update_texture_info(s->waveform.pipeline_compat, s->waveform.texture_info);
+        s->waveform.texture_image_rev = image->rev;
+    }
```

Comment:
can't we have the reference revision in the texture info so that `ngli_pipeline_compat_update_texture_info` decides whether to do the update or not?

Reply:
I did that originally but I found that holding this information in the shader crafter is fragile as:
1. this information does not belong to the shader crafter
2. it prevents the crafter from being re-used to create others pipelines.

Yes, having `ngli_pipeline_compat_update_texture_info()` handle that for us would have been great but I believe this needs to be done as part of a big revamp of pipeline_compat (and its goal).